build/SnippetExtractor/Program.cs

```diff
@@ -0,0 +1,220 @@
+            }
+        }
+
+        private async static Task<Document> RewriteInvocationsAsync(Document document)
+        {
+            ExpressionSyntax consoleWriteLineExpression =
+                SyntaxFactory.MemberAccessExpression(
+                    SyntaxKind.SimpleMemberAccessExpression,
+                    SyntaxFactory.IdentifierName("Console"),
+                    SyntaxFactory.IdentifierName("WriteLine"));
+            var model = await document.GetSemanticModelAsync();
+            var errors = model.GetDiagnostics().Where(d => d.Severity == DiagnosticSeverity.Error);
+            if (errors.Any())
+            {
+                throw new Exception("Snippet doesn't build:\r\n" + string.Join("\r\n", errors.Select(e => e.GetMessage())));
+            }
+
+            var assertType = model.Compilation.GetTypeByMetadataName("NUnit.Framework.Assert");
+            var snippetType = model.Compilation.GetTypeByMetadataName("NodaTime.Demo.Snippet");
+            var root = await document.GetSyntaxRootAsync();
+            return document.WithSyntaxRoot(root.ReplaceNodes(root.DescendantNodes().OfType<InvocationExpressionSyntax>(),
+                ReplaceNode));
+
+            SyntaxNode ReplaceNode(InvocationExpressionSyntax oldNode, InvocationExpressionSyntax newNode)
+            {
+                var symbol = model.GetSymbolInfo(oldNode).Symbol;
+                switch (symbol)
+                {
+                    case IMethodSymbol method when method.ContainingType == assertType && method.Name == "AreEqual":
+                        return newNode
+                            .WithExpression(consoleWriteLineExpression)
+                            .WithArgumentList(SyntaxFactory.ArgumentList().AddArguments(newNode.ArgumentList.Arguments[1]))
+                            .WithTriviaFrom(newNode);
+                    case IMethodSymbol method when method.ContainingType == assertType && method.Name == "True":
+                        return newNode
+                            .WithExpression(consoleWriteLineExpression)
+                            .WithArgumentList(SyntaxFactory.ArgumentList().AddArguments(newNode.ArgumentList.Arguments[0]))
+                            .WithTriviaFrom(newNode);
+                    case IMethodSymbol method when method.ContainingType == snippetType && method.Name == "For":
+                        return newNode.ArgumentList.Arguments[0].Expression;
+                    default: return newNode;
+                }
+            }
+        }
+
+        private async static Task<Document> RemoveUnusedImportsAsync(Document document)
+        {
+            // TODO: See if there's a better way of doing this.
+            var compilation = await document.Project.GetCompilationAsync();
+            var tree = await document.GetSyntaxTreeAsync();
+            var root = tree.GetRoot();
+            var unusedImportNodes = compilation.GetDiagnostics()
+                .Where(d => d.Id == "CS8019")
```

Comment:
Assuming you were being sarcsastic - I know, it's horrible. I believe there's a nice way of doing this by adding a "simplify" annotation and then asking Roslyn to simplify everything. But I haven't worked it out yet, and I thought I'd give @DustinCampbell a rest after Build...